src/org/opensolaris/opengrok/analysis/OGKTextField.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * This work is licensed under the Creative Commons Attribution-ShareAlike 3.0
+ * United States License. To view a copy of this license, visit
+ * http://creativecommons.org/licenses/by-sa/3.0/us/ or send a letter to
+ * Creative Commons, PO Box 1866, Mountain View, CA 94042, USA.
+ *
+ * Copyright (c) 2012, https://stackoverflow.com/users/1270457/amas
+ * Copyright (c) 2012, https://stackoverflow.com/questions/11945728/how-to-use-termvector-lucene-4-0
+ * Portions Copyright (c) 2018, Chris Fraire <cfraire@me.com>.
+ */
+
+package org.opensolaris.opengrok.analysis;
+
+import java.io.Reader;
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.FieldType;
+import org.apache.lucene.index.IndexOptions;
+
+/**
+ * Represents an OpenGrok-customized tokenized text field to centralize settings
+ * across all the analyzers.
+ */
+public class OGKTextField extends Field {
+
+    /** Indexed, tokenized, not stored. */
+    public static final FieldType TYPE_NOT_STORED = new FieldType();
+
+    /** Indexed, tokenized, stored. */
+    public static final FieldType TYPE_STORED = new FieldType();
+
+    static {
+        TYPE_NOT_STORED.setIndexOptions(
+            IndexOptions.DOCS_AND_FREQS_AND_POSITIONS_AND_OFFSETS);
+        TYPE_NOT_STORED.setTokenized(true);
+        TYPE_NOT_STORED.freeze();
+
+        TYPE_STORED.setIndexOptions(
+            IndexOptions.DOCS_AND_FREQS_AND_POSITIONS_AND_OFFSETS);
+        TYPE_STORED.setTokenized(true);
+        TYPE_STORED.setStored(true);
+        TYPE_STORED.freeze();
+    }
+
+    /**
+     * Creates a new un-stored instance with Reader value.
+     * @param name
```

Comment:
Could you please add comments for the parameters? I know that they are mostly self explaining so you could remove them if you prefer to (that's what I would do). It's just that if they have this format then they do not provide any additional info than the parameters already in code. :)

Reply:
ok removed